datafusion/Cargo.toml

```diff
@@ -54,6 +54,10 @@ arrow = { version = "^5.3", features = ["prettyprint"] }
 parquet = { version = "^5.3", features = ["arrow"] }
 sqlparser = "0.11"
 paste = "^1.0"
+# these two are for approx_distinct
+twox-hash = "1.6.1"
+pdatastructs = "0.6.0"
```

Comment:
`pdatastructs` author here: I should probably release a new version since it contains a few smaller fixes. Also if you have any trouble w/ this crate or need any features, feel free to ping me.

Reply:
I am all for reusing code if possible, but I wonder what you think of bringing any code needed from `pdatastructs` into the DataFusion crate itself?

I am thinking that given the goals of DataFusion as an embedded query engine as well as its active body of maintainers, we might better serve our users and ourselves by keeping the dependency chain smaller, especially for code that may be unlikely to change much

I also noticed that not much in crates.io seems to depend yet on `pdatastructs`  (which I take as a measure of their relative maturity):   https://crates.io/crates/pdatastructs/reverse_dependencies

On the other hand, perhaps starting to use it more actively will bring in more users to the `pdatastructs` community

Reply:
> think of bringing any code needed from pdatastructs into the DataFusion crate itself?

i think it's a good idea. @crepererum do you mind?

Reply:
Serde support can definitely be arranged (https://github.com/crepererum/pdatastructs.rs/issues/61).

Regarding the code move:
The goal of the pdatastructs project is to provide these kind of data structures (sketches as they're mostly called) to a wider audience, because I my view they are undervalued by developers and Rust is a good platform to implement them in a performant yet readable way. It is also for that reason that the documentation tries to explain quite some internals of the data structures. I don't mind if you copy code around, but I'm won't give up the original pdatastructs crate for that. With the latter I think it's unavoidable that the code will diverge on the long run. I you're worried about the dependency set pdatastructs itself, we can talk about that however (e.g. by using feature flags for the different structs or splitting it into smaller crates).